### PR TITLE
[BENCH-715] AWS S3 Storage folder Connected Test

### PR DESCRIPTION
**Pull request by dexamundsen**
- AWS S3 Storage folder Connected Test
- Add resource creation to cloud context connected test

image

### CODE REVIEW

File: service/src/test/java/bio/terra/workspace/common/BaseAwsConnectedTest.java [hunk 1]
```diff
@@ -23,7 +23,7 @@ public class BaseAwsConnectedTest extends BaseTest {
   @MockBean protected FeatureService featureService;
 
   @BeforeAll
-  public void setup() {
+  public void init() throws Exception {
```
Comment (yuhuyoyo) [inline service/src/test/java/bio/terra/workspace/common/BaseAwsConnectedTest.java:26]: is the throws neede?

**Review by yuhuyoyo — APPROVED** [review]

File: service/src/test/java/bio/terra/workspace/common/BaseAwsConnectedTest.java (hunk 1 above)
Comment (dexamundsen) [reply service/src/test/java/bio/terra/workspace/common/BaseAwsConnectedTest.java:26]: Yes, the overloaded definition in subclass needs this

File: service/src/test/java/bio/terra/workspace/common/fixtures/ControlledAwsResourceFixtures.java [hunk 2]
```diff
@@ -134,17 +135,27 @@ public static AwsCloudContext makeAwsCloudContext() {
       (DeleteObjectsResponse)
           DeleteObjectsResponse.builder().sdkHttpResponse(SDK_HTTP_RESPONSE_400).build();
 
+  public static String uniqueS3StorageFolderName(String resourceName) {
```
[the remark this answers is not in the document]
Comment (dexamundsen) [reply service/src/test/java/bio/terra/workspace/common/fixtures/ControlledAwsResourceFixtures.java:138]: Used this for readability (not against switching to appendRandNumber). I will switch over in the next PR

[the remark this answers is not in the document]
Comment (dexamundsen) [reply service/src/test/java/bio/terra/workspace/common/fixtures/ControlledAwsResourceFixtures.java:138]: tracking in https://verily.atlassian.net/browse/BENCH-319

**Pull request merged**